### ISSUE TITLE
No default :ex_cldr backend is configured

### ISSUE THREAD

**Issue by maennchen** [kipcole9/cldr_units#22]
## Issue

When using `MyApp.Cldr.Unit.to_string!(7.3, unit: :kilogram)`, the following error is raised.

```
** (Cldr.NoDefaultBackendError) No default :ex_cldr backend is configured
    (ex_cldr 2.22.0) lib/cldr/config/config.ex:332: Cldr.Config.default_backend/0
    (ex_cldr 2.22.0) lib/cldr.ex:614: Cldr.validate_locale/1
    (ex_cldr_units 3.5.2) lib/cldr/unit.ex:2050: Cldr.Unit.validate_grammatical_gender/2
    (ex_cldr_units 3.5.2) lib/cldr/unit/format.ex:298: Cldr.Unit.Format.normalize_options/2
    (ex_cldr_units 3.5.2) lib/cldr/unit/format.ex:193: Cldr.Unit.Format.to_string/3
    (ex_cldr_units 3.5.2) lib/cldr/unit/format.ex:281: Cldr.Unit.Format.to_string!/3
```

I've tried to use different notations and always get the same result:

```
iex> MyBluePlanet.ClimateActionsAdmin.Cldr.Unit.to_string!(7.3, unit: :kilogram)
** (Cldr.NoDefaultBackendError) No default :ex_cldr backend is configured
    (ex_cldr 2.22.0) lib/cldr/config/config.ex:332: Cldr.Config.default_backend/0
    (ex_cldr 2.22.0) lib/cldr.ex:614: Cldr.validate_locale/1
    (ex_cldr_units 3.5.2) lib/cldr/unit.ex:2050: Cldr.Unit.validate_grammatical_gender/2
    (ex_cldr_units 3.5.2) lib/cldr/unit/format.ex:298: Cldr.Unit.Format.normalize_options/2
    (ex_cldr_units 3.5.2) lib/cldr/unit/format.ex:193: Cldr.Unit.Format.to_string/3
    (ex_cldr_units 3.5.2) lib/cldr/unit/format.ex:281: Cldr.Unit.Format.to_string!/3
iex> Cldr.Unit.to_string!(7.3, MyBluePlanet.ClimateActionsAdmin.Cldr, unit: :kilogram)
** (Cldr.NoDefaultBackendError) No default :ex_cldr backend is configured
    (ex_cldr 2.22.0) lib/cldr/config/config.ex:332: Cldr.Config.default_backend/0
    (ex_cldr 2.22.0) lib/cldr.ex:614: Cldr.validate_locale/1
    (ex_cldr_units 3.5.2) lib/cldr/unit.ex:2050: Cldr.Unit.validate_grammatical_gender/2
    (ex_cldr_units 3.5.2) lib/cldr/unit/format.ex:298: Cldr.Unit.Format.normalize_options/2
    (ex_cldr_units 3.5.2) lib/cldr/unit/format.ex:193: Cldr.Unit.Format.to_string/3
    (ex_cldr_units 3.5.2) lib/cldr/unit/format.ex:281: Cldr.Unit.Format.to_string!/3
iex> MyBluePlanet.ClimateActionsAdmin.Cldr.Unit.to_string!(7.3, unit: :kilogram, backend: MyBluePlanet.ClimateActionsAdmin.Cldr)
** (Cldr.NoDefaultBackendError) No default :ex_cldr backend is configured
    (ex_cldr 2.22.0) lib/cldr/config/config.ex:332: Cldr.Config.default_backend/0
    (ex_cldr 2.22.0) lib/cldr.ex:614: Cldr.validate_locale/1
    (ex_cldr_units 3.5.2) lib/cldr/unit.ex:2050: Cldr.Unit.validate_grammatical_gender/2
    (ex_cldr_units 3.5.2) lib/cldr/unit/format.ex:298: Cldr.Unit.Format.normalize_options/2
    (ex_cldr_units 3.5.2) lib/cldr/unit/format.ex:193: Cldr.Unit.Format.to_string/3
    (ex_cldr_units 3.5.2) lib/cldr/unit/format.ex:281: Cldr.Unit.Format.to_string!/3
iex> Cldr.Unit.to_string!(7.3, unit: :kilogram, backend: MyBluePlanet.ClimateActionsAdmin.Cldr)
** (Cldr.NoDefaultBackendError) No default :ex_cldr backend is configured
    (ex_cldr 2.22.0) lib/cldr/config/config.ex:332: Cldr.Config.default_backend/0
    (ex_cldr 2.22.0) lib/cldr.ex:449: Cldr.put_default_locale/1
    (ex_cldr 2.22.0) lib/cldr.ex:395: Cldr.default_locale/0
    (ex_cldr_units 3.5.2) lib/cldr/unit/format.ex:153: Cldr.Unit.Format.to_string/3
    (ex_cldr_units 3.5.2) lib/cldr/unit/format.ex:281: Cldr.Unit.Format.to_string!/3
```

## Versions

```
mix deps | grep ex_cldr
* ex_cldr 2.22.0 (Hex package) (mix)
  locked at 2.22.0 (ex_cldr) c6c8aae8
* ex_cldr_calendars 1.14.1 (Hex package) (mix)
  locked at 1.14.1 (ex_cldr_calendars) ffd960e9
* ex_cldr_currencies 2.9.1 (Hex package) (mix)
  locked at 2.9.1 (ex_cldr_currencies) 1bc6f9ea
* ex_cldr_dates_times 2.7.2 (Hex package) (mix)
  locked at 2.7.2 (ex_cldr_dates_times) ae6bcb69
* ex_cldr_lists 2.8.0 (Hex package) (mix)
  locked at 2.8.0 (ex_cldr_lists) 455406d4
* ex_cldr_numbers 2.18.0 (Hex package) (mix)
  locked at 2.18.0 (ex_cldr_numbers) 9663db9c
* ex_cldr_units 3.5.2 (Hex package) (mix)
  locked at 3.5.2 (ex_cldr_units) edb4066f
```

**Comment by maennchen** [comment]
Thanks @kipcole9 :heart:

**Comment by kipcole9** [comment]
Thanks for the report, that was quite sloppy of me. The fix requires an update to both [ex_cldr version 2.22.1](https://hex.pm/packages/ex_cldr/2.22.1) and [ex_cldr_units version 3.5.3](https://hex.pm/packages/ex_cldr_units/3.5.3).

The changelog entries are:

### Ex_cldr Bug Fixes

* `Cldr.Number.PluralRule.plural_type/2` correcly returns a plural type when no backend is provided, no default backend is configured and the locale is a `t:Cldr.LanguageTag`

* `Cldr.validate_locale/1` doesn't raise if there is no default backend configured and the locale is a `t:Cldr.LanguageTag`

### Ex_cldr_units Bug Fixes

* Fixes formatting units when invoking `MyApp.Cldr.Unit.to_string/2` (ie on the backend module) and no default backend is configured. Thanks again to @maennchen. Closes #22.  Requires at least [ex_cldr version 2.22.1](https://hex.pm/packages/ex_cldr/2.22.1)
